lib/dl_api_lib_testing/dl_api_lib_testing/connector/dataset_suite.py

```diff
@@ -33,3 +38,23 @@ def test_remove_connection(
         sync_us_manager.delete(sync_us_manager.get_by_id(saved_connection_id))
         dataset_resp = control_api.load_dataset(saved_dataset)
         assert dataset_resp.status_code == 200, dataset_resp.json
+
+    def test_dataset__delete_result_field__available_after_refresh(
+        self,
+        saved_dataset: Dataset,
+        control_api: SyncHttpDatasetApiV1,
+    ) -> None:
+        # arrange
+        dataset = saved_dataset
+        result_field_to_remove = next(iter(dataset.result_schema))
+
+        # act
+        dataset = control_api.apply_updates(dataset, updates=[result_field_to_remove.delete()]).dataset
+        dataset_after_deletion = control_api.save_dataset(dataset).dataset
+        dataset_after_reload = control_api.refresh_dataset_sources(
+            dataset_after_deletion, dataset.sources._item_ids
+        ).dataset
+
+        # assert
```

Comment:
let's also assert the number of fields compared to the original dataset

Reply:
done